### PR TITLE
Avoid division by zero (for issue #708)

### PR DESCRIPTION
**Pull request by Buddhima**
File: libraries/joomla/image/image.php
Method: prepareDimensions()

Issue: If zero is given for the one of the arguments $height or $width, following section will produce an error.
Solution: Treat 3 instances where $height=0,$width=0 and both not equals zero

Signed-off-by: Buddhima Wijeweera

### CODE REVIEW

**Comment by eddieajau** [comment]
There are a lot of style errors in this file.  Have a look at this report:

http://developer.joomla.org/pulls/logs/1095checkstyle.xml

Search for the following line:

<file name=".../libraries/joomla/image/image.php">

Any error that is not complaining about an underscore needs to be fixed.  Thanks.

File: libraries/joomla/image/image.php [hunk 1]
```diff
@@ -671,9 +671,30 @@ protected function prepareDimensions($width, $height, $scaleMethod)
...
+                            
+                            // To avoid dividing by $height if it is zero                            
+                            elseif ($width!=0 && $height==0) {
+                               $rx = $this->getWidth() / $width;
+				$ry = $this->getHeight() ; 
+                            }
+                            
+                            // Both $height and $width are zero
+                            elseif($width==0 && $height==0){
+                                $rx = $this->getWidth() / $width;
```
Comment (benjaminpick) [inline libraries/joomla/image/image.php]: This cannot be correct: $width is 0, so this will divide by 0.

**Comment by benjaminpick** [comment]
Also, it would be good to explain in the Doc-Header what a zero value means.

**Comment by Buddhima** [comment]
Hi,

What I thought regarding this matter was there should be a value for rx when width =0. And I know when values getting closer to 0 , rx should tends to infinity. Therefore I can't specifically say a value. Then I thought of as a standard, assign rx with the value given by getWidth(). Same thing goes for ry too. 
But I got to know that throwing exceptions from classes like this is allowed.

So my new solution would be this.
Since this this method consider 2 cases for SCALE_INSIDE and SCALE_OUTSIDE there mustn't be a situation where user is given a chance of entering values less than 1. 
Anything designing on top of this should obey that rule. If this zero values are coming from an application, it should be a mistake done by developer.
Another case might be issues with web browser. Following links [1], [2] would show that.

My solution is to throw exceptions in those places where height or  width would be zero.

[1] http://name1price.com/how-to-get-customers-with-singapore-web-design-basic/jquery-tutorials/149-jquery-get-image-height-or-width-returns-0-problem-.html
[2] https://getsatisfaction.com/galleria/topics/could_not_extract_width_height_from_image_in_webkit_browsers 

Special thank goes to Elin for providing necessary guidance.

File: libraries/joomla/image/image.php [hunk 2]
```diff
@@ -654,6 +654,8 @@ protected function getFilterInstance($type)
 	 *
 	 * @return  object
 	 *
+         * @throws  LogicException          If width,height or both given as zero
```
Comment (eddieajau) [inline libraries/joomla/image/image.php]: I think you have space indenting on these two lines.

File: libraries/joomla/image/image.php [hunk 3]
```diff
@@ -671,9 +673,20 @@ protected function prepareDimensions($width, $height, $scaleMethod)
 
 			case self::SCALE_INSIDE:
 			case self::SCALE_OUTSIDE:
-				$rx = $this->getWidth() / $width;
-				$ry = $this->getHeight() / $height;
-
+			
+				// Both $height or $width cannot be zero
+				if($width == 0 || $height == 0)
```
Comment (eddieajau) [inline libraries/joomla/image/image.php]: Space needed after the `if`.

File: libraries/joomla/image/image.php [hunk 4]
```diff
@@ -671,9 +673,20 @@ protected function prepareDimensions($width, $height, $scaleMethod)
...
+				{
+					throw new LogicException(' Both height or width cannot be zero ');
```
Comment (eddieajau) [inline libraries/joomla/image/image.php]: This should be an InvalidArgumentException

```
throw new InvalidArgumentException('Width or height cannot be zero with this scale method.');
```

Also adjust the DocBlock.  Thanks.

Comment (Buddhima) [inline libraries/joomla/image/image.php]: thanks  eddieajau ,
Corrected all mistakes and committed !

**Comment by elinw** [comment]
The pull tester says you have white space on three lines.

http://developer.joomla.org/pulls/pulls/1095.html

**Pull request merged**